test/dtypes/test_uintx.py

```diff
@@ -6,7 +6,10 @@
 
 from torchao.dtypes.uintx.Uintx import to_uintx
 from torchao.quantization.quant_api import quantize_, uintx_weight_only
-from torchao.utils import TORCH_VERSION_AFTER_2_5
+from torchao.utils import (
+    TORCH_VERSION_AFTER_2_3,
```

Comment:
This import makes sense, but also is kind of useless since tests don't run unless Torch version is nightly. Maybe we can change tests to running if the version is after 2.3, and only calling `torch.compile()` if the torch version is nightly.

Reply:
sorry I don't quite follow, this is to guard against reference to `torch.uintx`, which is added in torch 2.3, so the tests will only run if torch is 2.3+

are you saying to add the version guard for tests as well?

also I didn't see a call for `torch.compile` in the test, are you saying we should add that

Reply:
Oh interesting, I didn't catch that we weren't using torch compile. That is intended usage so it should be tested.

 I was trying to say the guard for version 2.5 was for torch compile since the nested sub-classes issue was fixed only recently. Currently tests only run if version after 2.5 so the version after 2.3 guard is not really doing anything. 
 
 Now I see it guards the import which is used in a statement before the tests are executed so it is still needed